smalltalksrc/VMMaker/DruidJIT.class.st

```diff
@@ -5153,79 +5149,80 @@ DruidJIT >> gen_PrimitiveNewWithArg [
 	self MoveR: Extra3Reg Mw: 16 r: SPReg.
 	self MoveR: Extra0Reg R: Extra3Reg.
 	self MoveR: Extra3Reg Mw: 24 r: SPReg.
+	self MoveR: Extra2Reg R: Extra1Reg.
 	currentBlock := self Label.
 	jump7 jmpTarget: currentBlock.
 	self MoveMw: 0 r: SPReg R: Extra3Reg.
-	self MoveR: Extra3Reg R: SendNumArgsReg.
+	self MoveR: Extra3Reg R: Extra2Reg.
 	s172 := 0.
 	self MoveMw: 8 r: SPReg R: Extra3Reg.
-	self MoveR: Extra3Reg R: ClassReg.
-	self CmpCq: 16rFFFF R: ClassReg.
+	self MoveR: Extra3Reg R: SendNumArgsReg.
+	self CmpCq: 16rFFFF R: SendNumArgsReg.
 	jumpTrue := self JumpAbove: 0.
-	self MoveCq: 0 R: ClassReg.
+	self MoveCq: 0 R: SendNumArgsReg.
 	jumpNext := self Jump: 0.
 	jumpTrue jmpTarget: self Label.
-	self MoveCq: 1 R: ClassReg.
+	self MoveCq: 1 R: SendNumArgsReg.
 	jumpNext jmpTarget: self Label.
 	self MoveMw: 8 r: SPReg R: Extra3Reg.
 	self CmpCq: 16rFFFF R: Extra3Reg.
 	jump7 := self JumpBelowOrEqual: 0.
 	jump10 := self Jump: 0.
 	currentBlock := self Label.
 	jump7 jmpTarget: currentBlock.
-	self MoveCq: s172 R: ClassReg.
+	self MoveCq: s172 R: SendNumArgsReg.
 	currentBlock := self Label.
 	jump10 jmpTarget: currentBlock.
-	self CmpCq: 1 R: ClassReg.
+	self CmpCq: 1 R: SendNumArgsReg.
 	jump10 := self JumpNonZero: 0.
 	self MoveMw: 8 r: SPReg R: Extra3Reg.
 	self CmpCq: 16r10000000000 R: Extra3Reg.
 	jump7 := self JumpBelowOrEqual: 0.
-	self MoveR: SendNumArgsReg R: ReceiverResultReg.
+	self MoveR: Extra2Reg R: ReceiverResultReg.
 	jump12 := self Jump: 0.
 	currentBlock := self Label.
 	jump7 jmpTarget: currentBlock.
-	self MoveR: SendNumArgsReg R: ReceiverResultReg.
+	self MoveR: Extra2Reg R: ReceiverResultReg.
 	jump7 := self Jump: 0.
 	currentBlock := self Label.
 	jump10 jmpTarget: currentBlock.
 	self MoveMw: 8 r: SPReg R: Extra3Reg.
 	self CmpCq: 255 R: Extra3Reg.
 	jump10 := self JumpBelow: 0.
-	self MoveR: SendNumArgsReg R: ReceiverResultReg.
+	self MoveR: Extra2Reg R: ReceiverResultReg.
 	jump6 := self Jump: 0.
 	currentBlock := self Label.
 	jump10 jmpTarget: currentBlock.
 	self MoveMw: 8 r: SPReg R: Extra3Reg.
 	self CmpCq: 1 R: Extra3Reg.
 	jump10 := self JumpAboveOrEqual: 0.
 	s191 := 8.
-	self MoveCq: s191 R: ClassReg.
+	self MoveCq: s191 R: SendNumArgsReg.
 	jump5 := self Jump: 0.
 	currentBlock := self Label.
 	jump10 jmpTarget: currentBlock.
 	s194 := 8.
-	self MoveCq: s194 R: ClassReg.
+	self MoveCq: s194 R: SendNumArgsReg.
 	self MoveMw: 8 r: SPReg R: Extra3Reg.
-	self MoveR: Extra3Reg R: Extra1Reg.
-	self MulR: ClassReg R: Extra1Reg.
-	self MoveR: Extra1Reg R: ClassReg.
+	self MoveR: Extra3Reg R: ClassReg.
+	self MulR: SendNumArgsReg R: ClassReg.
+	self MoveR: ClassReg R: SendNumArgsReg.
```

Comment:
Here we have a multiply by 8 :) => << 3

Reply:
https://github.com/Alamvic/druid/issues/107